### PR TITLE
Add Settings UI

### LINKED ISSUE #62: Add settings UI

### PR DESCRIPTION
**Pull request by tobibodamer**
Closes #62

### CODE REVIEW

File: H2MLauncher.UI/ViewModels/SettingsViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    [ObservableProperty]
+    private string _iw4mMasterServerUrl = "";
+
+    public ICommand ApplyCommand { get; set; }
+
+    public ICommand CancelCommand { get; set; }
+
+    public SettingsViewModel(IWritableOptions<H2MLauncherSettings> options)
+    {
+        MwrLocation = options.Value.MWRLocation;
```
Comment (metalglove) [inline H2MLauncher.UI/ViewModels/SettingsViewModel.cs:30]: Check for nulls

File: H2MLauncher.UI/appsettings.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "H2MLauncher": {
-    "MWRLocation": "weweewew"
```
Comment (metalglove) [inline H2MLauncher.UI/appsettings.json:3]: Why did you remove MWRLocation?

**Review by metalglove — CHANGES_REQUESTED** [review]

File: H2MLauncher.UI/ViewModels/SettingsViewModel.cs (hunk 1 above)
Comment (tobibodamer) [reply H2MLauncher.UI/ViewModels/SettingsViewModel.cs:30]: I cannot be null, defaults to empty string

Comment (metalglove) [reply H2MLauncher.UI/ViewModels/SettingsViewModel.cs:30]: I mean the option object

File: H2MLauncher.UI/appsettings.json (hunk 2 above)
Comment (tobibodamer) [reply H2MLauncher.UI/appsettings.json:3]: The default should be empty which is the current directory (relative path). When the game is first detected i set it to the full path if its empty in #58 (https://github.com/Bowhza/H2M-Launcher/pull/58/files#diff-a65fb504b4ad5b079ce84653e7c83bdb1feff480d9b85b6149972d4e0a5667edR87)

File: H2MLauncher.Core/Services/RaidMaxService.cs [hunk 3]
```diff
@@ -26,9 +27,15 @@ public async Task<List<RaidMaxServer>> GetServerInfosAsync(CancellationToken can
 
             try
             {
-                HttpResponseMessage response = await _httpClient.GetAsync(APILINK, cancellationToken);
-                response.EnsureSuccessStatusCode();
-                JsonSerializerOptions options = new JsonSerializerOptions();
+                string url = $"{_options.CurrentValue.IW4MMasterServerUrl.TrimEnd('/')}/instance";
```
Comment (metalglove) [inline H2MLauncher.Core/Services/RaidMaxService.cs:30]: use [trimendingdirectoryseparator](https://learn.microsoft.com/en-us/dotnet/api/system.io.path.trimendingdirectoryseparator?view=net-8.0)

File: H2MLauncher.UI/MainWindow.xaml.cs [hunk 4]
```diff
@@ -87,5 +87,16 @@ private void DataGridRow_GotFocus(object sender, RoutedEventArgs e)
...
             ((DataGridRow)sender).IsSelected = true;
         }
+
+        private void ComboBoxItem_Selected(object sender, RoutedEventArgs e)
+        {
+            ((ComboBoxItem)sender).IsSelected = false;
+            e.Handled = true;
+        }
+
+        private void ToggleButton_Checked(object sender, RoutedEventArgs e)
```
Comment (metalglove) [inline H2MLauncher.UI/MainWindow.xaml.cs:97]: `e.Handled` ? or forgot to remove?

**Review by metalglove — APPROVED** [review]
